Test/Integration/CarrierTestAbstract.php

```diff
@@ -0,0 +1,232 @@
+use Magento\TestFramework\Helper\Bootstrap;
+use StoreKeeper\StoreKeeper\Api\OrderApiClient;
+use StoreKeeper\StoreKeeper\Model\Carrier\Storekeeper as StorekeeperCarrier;
+
+abstract class CarrierTestAbstract extends \PHPUnit\Framework\TestCase
+{
+    protected $carrier = 'storekeeper';
+    protected $storekeeperCarrier;
+    protected $orderApiClientMock;
+    protected $scopeConfig;
+    protected $configHelper;
+    protected $storeManager;
+    protected $_rateResultFactory;
+    protected $_rateMethodFactory;
+    protected $shipping;
+    protected $logger;
+
+    protected function setUp(): void
+    {
+        $objectManager = new ObjectManager($this);
+        $this->orderApiClientMock = $this->createMock(OrderApiClient::class);
+        $this->scopeConfig = Bootstrap::getObjectManager()->create(\Magento\Framework\App\Config\ScopeConfigInterface::class);
+        $this->configHelper = Bootstrap::getObjectManager()->create(\StoreKeeper\StoreKeeper\Helper\Config::class);
+        $this->storeManager = Bootstrap::getObjectManager()->create(\Magento\Store\Model\StoreManagerInterface::class);
+        $this->_rateResultFactory = Bootstrap::getObjectManager()->create(\Magento\Shipping\Model\Rate\ResultFactory::class);
+        $this->_rateMethodFactory = Bootstrap::getObjectManager()->create(\Magento\Quote\Model\Quote\Address\RateResult\MethodFactory::class);
+        $this->logger = Bootstrap::getObjectManager()->create(\StoreKeeper\StoreKeeper\Logger\Logger::class);
+        $this->shipping = Bootstrap::getObjectManager()->create(Shipping::class);
+
+        $this->orderApiClientMock->method('getListShippingMethodsForHooks')->willReturn($this->getApiRates());
+
+        $this->storekeeperCarrier = $objectManager->getObject(StorekeeperCarrier::class, [
+            'orderApiClient' => $this->orderApiClientMock,
+            'scopeConfig' => $this->scopeConfig,
+            'configHelper' => $this->configHelper,
+            'storeManager' => $this->storeManager,
+            '_rateResultFactory' => $this->_rateResultFactory,
+            '_rateMethodFactory' => $this->_rateMethodFactory,
+            'logger' => $this->logger
+        ]);
+    }
+
+    /**
+     * Test that carrier available and applicable
+     *
+     * @return void
+     */
+    public function testCollectRatesWhenShippingCarrierIsAvailableAndApplicable()
+    {
+        $request = $this->createRequest();
+        $expectedRates = $this->getExpectedRates();
+
+        $actualRates = $this->storekeeperCarrier->collectRates($request)->getAllRates();
+
+        //Assert equal amount of expected and acual collected rates
+        self::assertEquals(count($expectedRates), count($actualRates));
+        foreach ($actualRates as $i => $actualRate) {
+            $actualRate = $actualRate->getData();
+            //Assert carrier code, title, method, method title, cost and price
+            self::assertEquals($expectedRates[$i], $actualRate);
+        }
```

Comment:
```suggestion
        foreach ($actualRates as $i => $actualRate) {
            $actualRatesData[$i] = $actualRate->getData();
        }
       self::assertEquals($expectedRates[$i], $actualRatesData);
```

Better to assert full array at once, than you also do not need the count assert above